tests/Nfield.Quota.Tests/QuotaFrameValidatorTests.cs

```diff
@@ -0,0 +1,435 @@
+﻿using System.Linq;
+using Nfield.Quota.Builders;
+using NUnit.Framework;
+
+namespace Nfield.Quota.Tests
+{
+    [TestFixture]
+    internal class QuotaFrameValidatorTests
+    {
+        [Test]
+        public void Definitions_UniqueIdsAndNames_HappyPath()
+        {
+            var quotaFrame = new QuotaFrameBuilder()
+                .Id("id")
+                .VariableDefinition("varId", "varName", "odinVarName", var =>
+                {
+                    var.Level("level1Id", "level1Name");
+                    var.Level("level2Id", "level2Name");
+                })
+                .Build();
+
+            var validator = new QuotaFrameValidator();
+            var result = validator.Validate(quotaFrame);
+
+            Assert.That(result.IsValid, Is.True);
+        }
+
+        [Test]
+        public void Definitions_CannotBeEmpty()
+        {
+            var quotaFrame = new QuotaFrame();
+
+            var validator = new QuotaFrameValidator();
+            var result = validator.Validate(quotaFrame);
+
+            Assert.That(result.Errors.Single().ErrorMessage,
+                Is.EqualTo("Quota frame definitions cannot be empty."));
+        }
+
+        [Test]
+        public void Definitions_EveryVariableNeedsAtLeastTwoLevels()
+        {
+            var quotaFrame = new QuotaFrameBuilder()
```

Comment:
Are we sure about the minimum of two, what if you are only interested in a certain trait (or not)?